packages/admin-panel-server/src/routes/mapOverlayVisualisations/ImportMapOverlayVisualisationRoute.ts

```diff
@@ -0,0 +1,199 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ *
+ */
+
+import assert from 'assert';
+import { Request, Response, NextFunction } from 'express';
+import fs from 'fs';
+
+import { Route } from '@tupaia/server-boilerplate';
+import { readJsonFile, reduceToDictionary, snakeKeys, UploadError, yup } from '@tupaia/utils';
+
+import { MeditrakConnection } from '../../connections';
+import {
+  MapOverlayVisualisationExtractor,
+  draftReportValidator,
+  mapOverlayGroupValidator,
+  draftMapOverlayValidator,
+  mapOverlayGroupRelationsValidator,
+} from '../../viz-builder';
+import type {
+  MapOverlayVizResource,
+  MapOverlayGroupRecord,
+  MapOverlayGroup,
+  MapOverlayGroupRelation,
+  MapOverlayGroupRelationRecord,
+} from '../../viz-builder';
+
+const importFileSchema = yup.object().shape(
+  {
+    mapOverlayGroups: yup.array().of(mapOverlayGroupValidator),
+    mapOverlayGroupRelations: yup.array().of(mapOverlayGroupRelationsValidator),
```

Comment:
```suggestion
    mapOverlayGroups: yup.array().of(mapOverlayGroupValidator.required()),
    mapOverlayGroupRelations: yup.array().of(mapOverlayGroupRelationsValidator.required()),
```
I think you probably want to use `.required()` here? Yup can be a bit unexpected with the default being that `undefined` is always supported unless otherwise specified using `.required()`. Currently this says it's an optional `array` of optional `mapOverlayGroups`/`mapOverGroupRelations`. We probably want optional `array` of required `mapOverlayGroups`/`mapOverGroupRelations` ?

Reply:
Sounds good, updated